crates/miden-testing/tests/integration/scripts/in_flight_swap.rs

```diff
@@ -0,0 +1,239 @@
+    assert!(bob.vault().assets().any(|asset| asset == offered_asset));
+}
+
+/// This is a modification of `create_swap_note` to create an in-flight swap note. The consumer of
+/// this note does not receive the `offered_asset` directly, and only acts as an intermediary. The
+/// consumer will create a new P2ID note with `sender` as target, containing the `requested_asset`.
+fn create_in_flight_swap_note(
+    sender_account_id: AccountId,
+    offered_asset: Asset,
+    requested_asset: Asset,
+) -> (Note, NoteDetails) {
+    let mut rng = RpoRandomCoin::new([Felt::new(1), Felt::new(2), Felt::new(3), Felt::new(4)]);
+    let note_type = NoteType::Public;
+    let aux = Felt::new(0);
+
+    let note_script =
+        NoteScript::compile(IN_FLIGHT_SWAP_NOTE_SCRIPT, TransactionKernel::testing_assembler())
+            .unwrap();
+
+    let payback_serial_num = rng.draw_word();
+    let payback_recipient = build_p2id_recipient(sender_account_id, payback_serial_num).unwrap();
+
+    let payback_recipient_word: Word = payback_recipient.digest().into();
+    let requested_asset_word: Word = requested_asset.into();
+    let payback_tag = NoteTag::from_account_id(sender_account_id);
+
+    let inputs = NoteInputs::new(vec![
+        payback_recipient_word[0],
+        payback_recipient_word[1],
+        payback_recipient_word[2],
+        payback_recipient_word[3],
+        requested_asset_word[0],
+        requested_asset_word[1],
+        requested_asset_word[2],
+        requested_asset_word[3],
+        payback_tag.into(),
+        NoteExecutionHint::always().into(),
+    ])
+    .unwrap();
+
+    let tag = build_swap_tag(note_type, &offered_asset, &requested_asset).unwrap();
+    let serial_num = rng.draw_word();
+
+    let metadata =
+        NoteMetadata::new(sender_account_id, note_type, tag, NoteExecutionHint::always(), aux)
+            .unwrap();
+    let assets = NoteAssets::new(vec![offered_asset]).unwrap();
+    let recipient = NoteRecipient::new(serial_num, note_script, inputs);
+    let note = Note::new(assets, metadata, recipient);
+
+    let payback_assets = NoteAssets::new(vec![requested_asset]).unwrap();
+    let payback_note = NoteDetails::new(payback_assets, payback_recipient);
+
+    (note, payback_note)
+}
+
+const IN_FLIGHT_SWAP_NOTE_SCRIPT: &str = r"
+use.miden::note
+use.miden::tx
+use.miden::contracts::wallets::basic
+use.miden::contracts::wallets::aux
+
+# CONSTANTS
+# =================================================================================================
+
+const.PRIVATE_NOTE=2
+
+#! Swap script:
+#! Creates a note consumable by note issuer containing requested ASSET.
+#!
+#! Requires that the account exposes:
+#! - miden::contracts::wallets::basic::create_note procedure.
+#! - miden::contracts::wallets::aux::add_asset_to_note procedure.
+#!
+#! Inputs:  []
+#! Outputs: []
+#!
+#! Note inputs are assumed to be as follows:
+#! - RECIPIENT
+#! - ASSET
+#! - TAG = [tag, 0, 0, 0]
```

Comment:
```suggestion
#! Note inputs are assumed to be as follows:
#! - RECIPIENT
#! - ASSET
#! - TAG = [tag, execution_hint, 0, 0]
```

Nit: We actually provide the `execution_hint` as well. I noticed the docs on the `SWAP.masm` note are also missing the `execution_hint`. Would you mind updating that as well?